mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,221 @@
+# Python Virtual Environments (venv's)
+
+## Introduction
+
+Python virtual environments (venv's) are a way to create isolated Python environments.
+You can install packages in a venv without affecting the system Python installation.
+Because a normal user cannot install packages globally, 
+venv's are a way to install packages without needing root access.
+This is useful when you need to use a package that is not available as a module on the HPC cluster.
+
+## Creating a venv
+
+A venv can be created with the following commands:
+
+```bash
+$ python -m venv myenv      # Create a new venv named myenv
+$ source myenv/bin/activate # Activate the venv
+```
+
+After activating the venv, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+It is now possible to run Python scripts that use the installed packages in the venv. To deactivate the venv, run:
+
+```bash
+$ deactivate
+```
+
+!!! note
+    Always prefer to use modules if they are available. 
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
+    ```
+
+## Combining venv's with modules
+
+You can combine packages installed in a venv with modules. The following script uses 
+pytorch (which is available as a module) and Poutyne (which is not available as a module):
+
+```python title="pytorch_poutine.py"
+import torch
+import poutyne
+
+...
+```
+
+We load a PyTorch package as a module and install Poutyne in a venv:
+
+```bash
+$ ml PyTorch/2.1.2-foss-2023a
```

Comment:
use `module load` instead of `ml`

Reply:
Fixed in [411c9d6](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/411c9d69a6b72d0f5b3b656d4a0cc2fc14e462d5)